### PR TITLE
Add Testing + Javadoc

### PR DESCRIPTION
**Pull request by Cheng20010201**
Add tesing & javadoc for my parts.

1. Luckily, there is no need to do GUI testing -> a lot of efforts are saved.
2. The tests are (surely) not comrehensive... But...

### CODE REVIEW

**Comment by Cheng20010201** [comment]
Sorry for the long PR but most of the 'code' are quite tedious.. :') Thanks!

File: src/test/java/seedu/weeblingo/model/FlashcardBookTest.java [hunk 1]
```diff
@@ -27,49 +40,52 @@ public void resetData_null_throwsNullPointerException() {
...
+    }
+
+    @Test
+    public void hasFlashcard_flashcardInAddressBook_returnsTrue() {
+        flashcardBook.addFlashcard(A_CARD);
+        assertTrue(flashcardBook.hasFlashcard(A));
+    }
+
+    @Test
+    public void hasFlashcard_flashcardWithSameIdentityFieldsInAddressBook_returnsTrue() {
```
Comment (xinweit) [inline src/test/java/seedu/weeblingo/model/FlashcardBookTest.java]: Might want to use flashcardbook for naming of all these tests

File: src/test/java/seedu/weeblingo/model/FlashcardBookTest.java [hunk 2]
```diff
@@ -27,49 +40,52 @@ public void resetData_null_throwsNullPointerException() {
...
-    //                .build();
-    //        assertTrue(flashcardBook.hasFlashcard(editedAlice));
-    //    }
+    @Test
+    public void hasFlashcard_flashcardNotInAddressBook_returnsFalse() {
+        assertFalse(flashcardBook.hasFlashcard(A_CARD));
+    }
+
+    @Test
+    public void hasFlashcard_flashcardInAddressBook_returnsTrue() {
```
Comment (xinweit) [inline src/test/java/seedu/weeblingo/model/FlashcardBookTest.java]: here

File: src/test/java/seedu/weeblingo/model/FlashcardBookTest.java [hunk 3]
```diff
@@ -27,49 +40,52 @@ public void resetData_null_throwsNullPointerException() {
...
-    //        Flashcard editedAlice = new FlashcardBuilder(ALICE).withAnswer(VALID_ANSWER_B)
-    //        .withTags(VALID_TAG_HUSBAND)
-    //                .build();
-    //        List<Flashcard> newFlashcards = Arrays.asList(ALICE, editedAlice);
-    //        FlashcardBookStub newData = new FlashcardBookStub(newFlashcards);
-    //
-    //        assertThrows(DuplicateFlashcardException.class, () -> flashcardBook.resetData(newData));
-    //    }
+    @Test
+    public void resetData_withValidReadOnlyAddressBook_replacesData() {
```
Comment (xinweit) [inline src/test/java/seedu/weeblingo/model/FlashcardBookTest.java]: ^

File: src/test/java/seedu/weeblingo/model/FlashcardBookTest.java [hunk 4]
```diff
@@ -27,49 +40,52 @@ public void resetData_null_throwsNullPointerException() {
...
-    //    @Test
-    //    public void hasFlashcard_flashcardWithSameIdentityFieldsInAddressBook_returnsTrue() {
-    //        flashcardBook.addFlashcard(ALICE);
-    //        Flashcard editedAlice = new FlashcardBuilder(ALICE).withAnswer(VALID_ANSWER_B)
-    //        .withTags(VALID_TAG_HUSBAND)
-    //                .build();
-    //        assertTrue(flashcardBook.hasFlashcard(editedAlice));
-    //    }
+    @Test
+    public void hasFlashcard_flashcardNotInAddressBook_returnsFalse() {
```
Comment (xinweit) [inline src/test/java/seedu/weeblingo/model/FlashcardBookTest.java]: ^

Comment (Cheng20010201) [reply src/test/java/seedu/weeblingo/model/FlashcardBookTest.java]: Thanks~

**Review by xinweit — APPROVED** [review]

**Pull request merged**